dataeng/jobs/analytics/WarehouseTransformsCIManual.groovy

```diff
@@ -0,0 +1,104 @@
+package analytics
+import static org.edx.jenkins.dsl.AnalyticsConstants.secure_scm
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_log_rotator
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_wrappers
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_publishers
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_triggers
+import static org.edx.jenkins.dsl.AnalyticsConstants.secure_scm_parameters
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_authorization
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.GHPRB_CANCEL_BUILDS_ON_UPDATE
+
+class WarehouseTransformsCIManual{
+    public static def job = { dslFactory, allVars ->
+        dslFactory.job("warehouse-transforms-ci-manual"){
+            authorization common_authorization(allVars)
+            logRotator common_log_rotator(allVars)
+            parameters secure_scm_parameters(allVars)
+            parameters {
+                stringParam('WAREHOUSE_TRANSFORMS_URL', allVars.get('WAREHOUSE_TRANSFORMS_URL'), 'URL for the warehouse-transforms repository.')
+                stringParam('WAREHOUSE_TRANSFORMS_BRANCH', allVars.get('WAREHOUSE_TRANSFORMS_BRANCH'), 'Must specify branch of warehouse-transforms repository to use.')
+                stringParam('GITHUB_PR_ID', allVars.get('GITHUB_PR_ID'), 'Must specify Github Pull Request Id')
+                stringParam('DBT_TARGET', allVars.get('DBT_TARGET'), 'DBT target from profiles.yml in analytics-secure.')
+                stringParam('DBT_PROFILE', allVars.get('DBT_PROFILE'), 'DBT profile from profiles.yml in analytics-secure.')
+                stringParam('DBT_PROJECT_PATH', allVars.get('DBT_PROJECT_PATH'), 'Path in warehouse-transforms to use as the dbt project, relative to "projects" (usually automated/applications or reporting).')
+                stringParam('RUN_DBT_TESTS_ONLY', allVars.get('RUN_DBT_TESTS_ONLY'), 'Mark it as false if you want to run dbt models')
+                stringParam('DBT_RUN_OPTIONS', allVars.get('DBT_RUN_OPTIONS'), 'Additional options to dbt run, such as --models for model selection. Details here: https://docs.getdbt.com/docs/model-selection-syntax')
+                stringParam('DBT_RUN_EXCLUDE', allVars.get('DBT_RUN_EXCLUDE'), 'Additional options to dbt run, such as --exclude. Details here: https://docs.getdbt.com/docs/model-selection-syntax')
+                stringParam('DBT_TEST_OPTIONS', allVars.get('DBT_TEST_OPTIONS'), 'Additional options to dbt test, such as --models for model selection. Details here: https://docs.getdbt.com/docs/model-selection-syntax')
+                stringParam('DBT_TEST_EXCLUDE', allVars.get('DBT_TEST_EXCLUDE'), 'Additional options to dbt test, such as --exclude. Details here: https://docs.getdbt.com/docs/model-selection-syntax')
```

Comment:
Can you clarify in the description if the user is expected to provide the `--<option>` flag?  Or just a hint, like:
```groovy
                 stringParam('DBT_TEST_EXCLUDE', allVars.get('DBT_TEST_EXCLUDE'), 'Additional options to dbt test, such as "--exclude <model>". Details here: https://docs.getdbt.com/docs/model-selection-syntax')
```
Alternatively, I've always wondered why we require parameters with CLI arguments in them.  Can't we just accept them as model selectors, and add the flag in the bash script later:

```bash
# If DBT_TEST_EXCLUDE is provided, prepend "--exclude "
DBT_TEST_EXCLUDE="${DBT_TEST_EXCLUDE:+--exclude }$DBT_TEST_EXCLUDE"
```

Reply:
Yes we can prepend it, but I feel it should be common across all jobs or at-least manual jobs

Reply:
Ok I just saw the dbt-manual job does not have DBT_TEST_OPTIONS or DBT_TEST_EXCLUDES parameters anymore :D
Question: What should I do now prepend the --exclude  or keep like as is ?

Reply:
I think this would be a good opportunity to cleanup the other two jobs to be consistent:

* dataeng/jobs/analytics/DBTManual.groovy
* dataeng/jobs/analytics/WarehouseTransformsCI.groovy

I think the `DBT_*_OPTIONS` can left alone, only `DBT_*_EXCLUDE` needs to be updated.  However, if you do decide to update `DBT_*_OPTIONS`, you should rename it to `DBT_*_MODELS` because it will no longer serve the purpose of passing arbitrary options.

I wouldn't normally be this pedantic, but I think this improves the UX of the various jobs, and my understanding is that data ops is tackling some quality-of-life improvements.  One time, confusion around how to format parameter values in jenkins jobs has caused partner support many lost hours of passing incorrectly formatted values to the event export jobs, and that's just one example.